lib/css_parser/rule_set.rb

```diff
@@ -22,90 +24,152 @@ class RuleSet
+      #   puts declarations['margin']
+      #   => #<CssParser::RuleSet::Declarations::Value:0x00000000030c1838 @important=true, @order=2, @value="0px auto">
+      #
+      # If the property already exists its value will be over-written.
+      # If the value is empty - property will be deleted
+      def []=(property, value)
+        property = normalize_property(property)
+
+        if value.is_a?(Value)
+          declarations[property] = value
+          return
+        end
+
+        if value.to_s.strip.empty?
+          delete(property)
+          return
+        end
+
+        declarations[property] = Value.new(value, order: @order += 1)
+      end
+      alias add_declaration! []=
+
+      def [](property)
+        declarations[normalize_property(property)]
+      end
+      alias get_value []
+
+      def key?(property)
+        declarations.key?(normalize_property(property))
+      end
+
+      def size
+        declarations.size
+      end
 
-      def initialize(value, important: nil, order: 0)
-        self.value = value
-        @important = important.nil? ? !value.match(CssParser::IMPORTANT_IN_PROPERTY_RX).nil? : important
-        @order = order
+      # Remove CSS declaration
+      # @param [#to_s] property property to be removed
+      #
+      # @example
+      #   declarations.delete('color')
+      def delete(property)
+        declarations.delete(normalize_property(property))
       end
+      alias remove_declaration! delete
 
-      def value=(value)
-        value = value.to_s.sub(/\s*;\s*\Z/, '').gsub(CssParser::IMPORTANT_IN_PROPERTY_RX, '').strip
-        raise ArgumentError, 'value is empty' if value.empty?
+      def each(&block)
+        declarations.sort_by { |_name, value| value.order }.each(&block)
+      end
 
-        @value = value.freeze
+      def to_s(options = {})
+        str = declarations.reduce(String.new) do |memo, (prop, value)|
+          importance = options[:force_important] || value.important ? ' !important' : ''
+          memo << "#{prop}: #{value.value}#{importance}; "
+        end
+        # TODO: Clean-up regexp doesn't seem to work
+        str.gsub!(/^[\s^({)]+|[\n\r\f\t]*|\s+$/mx, '')
+        str.strip!
+        str
       end
 
-      def to_s
-        return value unless important
+    private
+
+      attr_reader :declarations
 
-        "#{value} !important"
+      def normalize_property(property)
+        property = property.downcase
+        property.strip!
+        property
       end
     end
 
+    extend Forwardable
+
     # Array of selector strings.
     attr_reader :selectors
 
     # Integer with the specificity to use for this RuleSet.
     attr_accessor :specificity
 
+    # @!method add_declaration!
+    #   @see CssParser::RuleSet::Declarations#add_declaration!
+    # @!method delete
+    #   @see CssParser::RuleSet::Declarations#delete
+    def_delegators :declarations, :add_declaration!, :delete
+    alias []= add_declaration!
+    alias remove_declaration! delete
```

Comment:
might be simpler to just define these 3 methods instead of dealing with aliases and forwardable